### ISSUE TITLE
Error with py.test .

### ISSUE THREAD

**Issue by hongliy** [usgs/shakemap#1114]
Please describe your issue here:

Hello,

I am having error of testing ShakeMap 4. The installation step did appear correct. I did try git pull and reinstall.

The following is the error message.

---
==================================== ERRORS ====================================
____________ ERROR collecting tests/shakemap/coremods/dyfi_test.py _____________
ImportError while importing test module '/Users/liliye/shakemap_src/tests/shakemap/coremods/dyfi_test.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
../opt/anaconda3/envs/shakemap/lib/python3.8/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests/shakemap/coremods/dyfi_test.py:9: in <module>
    from shakemap.coremods.dyfi import _get_dyfi_dataframe
shakemap/coremods/dyfi.py:9: in <module>
    from impactutils.io.table import dataframe_to_xml
E   ModuleNotFoundError: No module named 'impactutils.io.table'
-------- generated xml file: /Users/liliye/shakemap_src/test-output.xml --------
=========================== short test summary info ============================
ERROR tests/shakemap/coremods/dyfi_test.py
!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!
=============================== 1 error in 1.90s ===============================

**Comment by emthompson-usgs** [comment]
I think there are two possible reasons for this. Either the install process failed for some reason, or you did not activate the "shakemap" virtual environment. I cannot tell which is the case from what you sent. Be sure to type
`conda activate shakemap`
and then
`conda list`
And send us the output.

**Comment by hongliy** [comment]
Please see the output as following:

`Last login: Thu Jan 21 20:58:24 on ttys000
(base) liliye@liliyes-MacBook-Air ~ % conda activate shakemap 
(shakemap) liliye@liliyes-MacBook-Air ~ % conda list
# packages in environment at /Users/liliye/opt/anaconda3/envs/shakemap:
#
# Name                    Version                   Build  Channel
affine                    2.3.0                      py_0  
alabaster                 0.7.12                     py_0  
appnope                   0.1.2           py38hecd8cb5_1001  
attrs                     20.3.0             pyhd3eb1b0_0  
autopep8                  1.5.4                      py_0  
babel                     2.9.0              pyhd3eb1b0_0  
backcall                  0.2.0                      py_0  
bcrypt                    3.2.0            py38haf1e3a3_0  
beautifulsoup4            4.9.3              pyha847dfd_0  
blas                      1.0                         mkl  
boost-cpp                 1.74.0               h43a636a_2    conda-forge
brotlipy                  0.7.0           py38h9ed2024_1003  
bzip2                     1.0.8                h1de35cc_0  
c-ares                    1.17.1               h9ed2024_0  
c-compiler                1.1.3                h35c211d_0    conda-forge
ca-certificates           2021.1.19            hecd8cb5_0  
cairo                     1.16.0            h4ec454a_1007    conda-forge
cartopy                   0.18.0           py38h2f0d74d_7    conda-forge
cctools                   949.0.1             hd9211c8_19    conda-forge
cctools_osx-64            949.0.1             h2f0f38f_19    conda-forge
certifi                   2020.12.5        py38hecd8cb5_0  
cffi                      1.14.4           py38h2125817_0  
cfitsio                   3.470                h01dc385_7    conda-forge
cftime                    1.3.1            py38he3068b8_0  
chardet                   3.0.4                    pypi_0    pypi
clang                     11.0.0               h694c41f_2    conda-forge
clang-11                  11.0.0          default_h9e6edd0_2    conda-forge
clang_osx-64              11.0.0              hb91bd55_10    conda-forge
clangxx                   11.0.0          default_h9e6edd0_2    conda-forge
click                     7.1.2              pyhd3eb1b0_0  
click-plugins             1.1.1                      py_0  
cligj                     0.7.1            py38hecd8cb5_0  
cloudpickle               1.6.0                      py_0  
colorama                  0.4.4              pyhd3eb1b0_0  
commonmark                0.9.1                      py_0  
compiler-rt               11.0.0               h01488ec_2    conda-forge
compiler-rt_osx-64        11.0.0               hd3c4e95_2    conda-forge
configobj                 5.0.6                    py38_1  
coverage                  5.3.1            py38h9ed2024_2  
cryptography              3.3.1            py38hbcfaee0_0  
curl                      7.71.1               hcb81553_8    conda-forge
cycler                    0.10.0                   py38_0  
cython                    0.29.21          py38h23ab428_0  
cytoolz                   0.11.0           py38haf1e3a3_0  
dask-core                 2021.1.0           pyhd3eb1b0_0  
decorator                 4.4.2                      py_0  
defusedxml                0.6.0                      py_0  
descartes                 1.1.0                      py_4  
django                    2.2.17                   pypi_0    pypi
docutils                  0.14                     pypi_0    pypi
et_xmlfile                1.0.1                   py_1001  
expat                     2.2.10               hb1e8313_2  
fiona                     1.8.18           py38hd34e6ff_0    conda-forge
flake8                    3.8.4                      py_0  
fontconfig                2.13.1            hd23ceaa_1004    conda-forge
freetype                  2.10.4               ha233b18_0  
freexl                    1.0.6                h9ed2024_0  
future                    0.18.2                   py38_1  
gdal                      3.1.4            py38hcc4f518_2    conda-forge
geos                      3.8.1                hb1e8313_0  
geotiff                   1.6.0                hc1004f4_3    conda-forge
gettext                   0.19.8.1             hb0f4f8b_2  
giflib                    5.2.1                haf1e3a3_0  
h5py                      2.10.0           py38h0601b69_1  
hdf4                      4.2.13               h39711bb_2  
hdf5                      1.10.6          nompi_hc5d9132_1114    conda-forge
icu                       68.1                 h74dc148_0    conda-forge
idna                      2.10               pyhd3eb1b0_0  
imageio                   2.9.0                      py_0  
imagesize                 1.2.0                      py_0  
impactutils               0.8.25             pyhd8ed1ab_2    conda-forge
importlib-metadata        2.0.0                      py_1  
iniconfig                 1.1.1              pyhd3eb1b0_0  
intel-openmp              2019.4                      233  
ipython                   7.19.0           py38h01d92e1_1  
ipython_genutils          0.2.0              pyhd3eb1b0_1  
jdcal                     1.4.1                      py_0  
jedi                      0.17.2           py38hecd8cb5_1  
jinja2                    2.11.2             pyhd3eb1b0_0  
jpeg                      9d                   hbcb3906_0    conda-forge
json-c                    0.13.1               h3efe00b_0  
kealib                    1.4.14               h82515a5_0    conda-forge
kiwisolver                1.3.0            py38h23ab428_0  
krb5                      1.17.1               hddcf347_0  
lcms2                     2.11                 h92f6f08_0  
ld64                      530                 hd2e7500_19    conda-forge
ld64_osx-64               530                 hea264c1_19    conda-forge
ldid                      2.1.2                h7660a38_2    conda-forge
libblas                   3.9.0           1_h508aa58_netlib    conda-forge
libcblas                  3.9.0           3_h3cd88c1_netlib    conda-forge
libclang-cpp11            11.0.0          default_h9e6edd0_2    conda-forge
libcomcat                 2.0.7              pyhd8ed1ab_2    conda-forge
libcurl                   7.71.1               h9bf37e3_8    conda-forge
libcxx                    11.0.1               habf9029_0    conda-forge
libdap4                   3.20.6               h993cace_1    conda-forge
libedit                   3.1.20191231         h1de35cc_1  
libev                     4.33                 haf1e3a3_0  
libffi                    3.3                  hb1e8313_2  
libgdal                   3.1.4                hbb49e5d_2    conda-forge
libgfortran               5.0.0           9_3_0_h6c81a4c_16    conda-forge
libgfortran5              9.3.0               h6c81a4c_16    conda-forge
libglib                   2.66.4               h7424822_1    conda-forge
libiconv                  1.16                 h1de35cc_0  
libkml                    1.3.0             h8ca2c65_1012    conda-forge
liblapack                 3.9.0           3_h3cd88c1_netlib    conda-forge
libllvm11                 11.0.0               hf85e3d2_0    conda-forge
libnetcdf                 4.7.4           nompi_h9d8a93f_107    conda-forge
libnghttp2                1.41.0               h7580e61_2  
libpng                    1.6.37               ha441bb4_0  
libpq                     12.3                 h7049927_3    conda-forge
libsodium                 1.0.18               h1de35cc_0  
libspatialite             5.0.0                hc85de81_0    conda-forge
libssh2                   1.9.0                ha12b0ac_1  
libtiff                   4.2.0                h355d032_0    conda-forge
libwebp-base              1.1.0                haf1e3a3_3  
libxml2                   2.9.10               h802ca4f_3    conda-forge
libxslt                   1.1.33               h5739fc3_2    conda-forge
llvm-openmp               8.0.1                h770b8ee_0    conda-forge
llvm-tools                11.0.0               hf85e3d2_0    conda-forge
lockfile                  0.12.2                   py38_0  
lxml                      4.5.1            py38h63b7cb6_0  
lz4-c                     1.9.3                h23ab428_0  
mapio                     0.7.26                     py_2    conda-forge
markupsafe                1.1.1            py38h1de35cc_1  
matplotlib                3.1.3                    pypi_0    pypi
mccabe                    0.6.1                    py38_1  
mkl                       2019.4                      233  
mkl-service               2.3.0            py38h9ed2024_0  
mkl_fft                   1.2.0            py38hc64f4ea_0  
mkl_random                1.1.1            py38h959d312_0  
more-itertools            8.6.0              pyhd3eb1b0_0  
multidict                 5.1.0            py38h9ed2024_2  
munch                     2.5.0                      py_0  
ncurses                   6.2                  h0a44026_1  
netcdf4                   1.5.5.1         nompi_py38h0bc7383_101    conda-forge
networkx                  2.5                        py_0  
numpy                     1.18.5                   pypi_0    pypi
obspy                     1.2.2            py38h65ad66c_0    conda-forge
olefile                   0.46                       py_0  
openjpeg                  2.4.0                h6cbf5cd_0    conda-forge
openmp                    8.0.1                         0    conda-forge
openpyxl                  3.0.6              pyhd3eb1b0_0  
openquake-engine          3.11.0                   pypi_0    pypi
openssl                   1.1.1i               h9ed2024_0  
packaging                 20.8               pyhd3eb1b0_0  
pandas                    1.1.5                    pypi_0    pypi
paramiko                  2.7.2                      py_0  
parso                     0.7.0                      py_0  
pcre                      8.44                 hb1e8313_0  
pexpect                   4.8.0              pyhd3eb1b0_3  
pickleshare               0.7.5           pyhd3eb1b0_1003  
pillow                    8.1.0            py38h5270095_0  
pip                       20.3.3           py38hecd8cb5_0  
pixman                    0.40.0               haf1e3a3_0  
pluggy                    0.13.1                   py38_0  
poppler                   0.89.0               hd735947_5    conda-forge
poppler-data              0.4.10               hecd8cb5_0  
postgresql                12.3                 he852019_3    conda-forge
proj                      7.1.1                h45baca5_3    conda-forge
prompt-toolkit            3.0.8                      py_0  
ps2ff                     1.5.2                      py_1    conda-forge
psutil                    5.6.7                    pypi_0    pypi
ptyprocess                0.7.0              pyhd3eb1b0_2  
py                        1.10.0             pyhd3eb1b0_0  
pycodestyle               2.6.0                      py_0  
pycparser                 2.20                       py_2  
pycrypto                  2.6.1           py38haf1e3a3_10  
pyflakes                  2.2.0                      py_0  
pygments                  2.7.4              pyhd3eb1b0_0  
pynacl                    1.4.0            py38haf1e3a3_1  
pyopenssl                 20.0.1             pyhd3eb1b0_1  
pyparsing                 2.4.7              pyhd3eb1b0_0  
pyproj                    2.6.1.post1      py38h4367d1d_3    conda-forge
pyshp                     1.2.3                    pypi_0    pypi
pysocks                   1.7.1                    py38_1  
pytest                    6.2.1            py38hecd8cb5_0  
pytest-azurepipelines     0.8.0                      py_0  
pytest-cov                2.11.1             pyhd3eb1b0_0  
pytest-faulthandler       2.0.1                      py_0    conda-forge
python                    3.8.5                h26836e1_1  
python-daemon             2.2.4                    py38_1  
python-dateutil           2.8.1                      py_0  
python_abi                3.8                      1_cp38    conda-forge
pytz                      2020.5             pyhd3eb1b0_0  
pywavelets                1.1.1            py38haf1e3a3_2  
pyyaml                    5.4              py38h9ed2024_1  
pyzmq                     19.0.2                   pypi_0    pypi
rasterio                  1.1.8            py38h287b4b6_0    conda-forge
readline                  8.0                  h1de35cc_0  
requests                  2.23.0                   pypi_0    pypi
rope                      0.18.0                     py_0  
scikit-image              0.16.2           py38h6c726b0_0  
scipy                     1.4.1                    pypi_0    pypi
setuptools                51.3.3           py38hecd8cb5_4  
shakemap                  4.0.2+424.ga51e4f60           dev_0    <develop>
shapely                   1.7.1            py38h8918236_1    conda-forge
simplekml                 1.3.5              pyh9f0ad1d_0    conda-forge
six                       1.15.0           py38hecd8cb5_0  
snowballstemmer           2.0.0                      py_0  
snuggs                    1.4.7                      py_0  
soupsieve                 2.1                pyhd3eb1b0_0  
sphinx                    3.4.3              pyhd3eb1b0_0  
sphinx-argparse           0.2.5              pyhd8ed1ab_1    conda-forge
sphinxcontrib-applehelp   1.0.2                      py_0  
sphinxcontrib-devhelp     1.0.2                      py_0  
sphinxcontrib-htmlhelp    1.0.3                      py_0  
sphinxcontrib-jsmath      1.0.1                      py_0  
sphinxcontrib-qthelp      1.0.3                      py_0  
sphinxcontrib-serializinghtml 1.1.4                      py_0  
sqlalchemy                1.3.21           py38h9ed2024_0  
sqlite                    3.34.0               h17101e1_0    conda-forge
sqlparse                  0.4.1                    pypi_0    pypi
strec                     2.1.7                      py_1    conda-forge
tapi                      1100.0.11            h9ce4665_0    conda-forge
tiledb                    2.1.6                h6a9efa5_1    conda-forge
tk                        8.6.10               hb0a8c7a_0  
toml                      0.10.1                     py_0  
toolz                     0.11.1             pyhd3eb1b0_0  
tornado                   6.1              py38h9ed2024_0  
traitlets                 5.0.5                      py_0  
tzcode                    2020f                h35c211d_0    conda-forge
urllib3                   1.25.11                  pypi_0    pypi
vcrpy                     4.1.1                    py38_0  
versioneer                0.19               pyhd8ed1ab_0    conda-forge
wcwidth                   0.2.5                      py_0  
wheel                     0.36.2             pyhd3eb1b0_0  
wrapt                     1.12.1           py38haf1e3a3_1  
xerces-c                  3.2.3                hd3a8c1e_2    conda-forge
xlrd                      2.0.1              pyhd3eb1b0_0  
xlsxwriter                1.3.7                      py_0  
xlwt                      1.3.0                    py38_0  
xz                        5.2.5                h1de35cc_0  
yaml                      0.2.5                haf1e3a3_0  
yapf                      0.30.0                     py_0  
yarl                      1.6.3            py38h9ed2024_0  
zipp                      3.4.0              pyhd3eb1b0_0  
zlib                      1.2.11               h1de35cc_3  
zstd                      1.4.8                hf387650_1    conda-forge
(shakemap) liliye@liliyes-MacBook-Air ~ % 
`

**Comment by hongliy** [comment]
I run the code in shakemap environment. It displayed the same type error



(shakemap) liliye@liliyes-MacBook-Air shakemap_src % pwd
/Users/liliye/shakemap_src
(shakemap) liliye@liliyes-MacBook-Air shakemap_src % py.test .
============================= test session starts ==============================
platform darwin -- Python 3.8.5, pytest-6.2.1, py-1.10.0, pluggy-0.13.1
rootdir: /Users/liliye/shakemap_src, configfile: pytest.ini
plugins: azurepipelines-0.8.0, cov-2.11.1
collected 134 items / 1 error / 133 selected                                   
##vso[results.publish type=JUnit;runTitle='Pytest results';]/Users/liliye/shakemap_src/test-output.xml
##vso[task.logissue type=error;]1 test(s) failed, 134 test(s) collected.
##vso[task.logissue type=warning;]Coverage XML was not created, skipping upload.

==================================== ERRORS ====================================
____________ ERROR collecting tests/shakemap/coremods/dyfi_test.py _____________
ImportError while importing test module '/Users/liliye/shakemap_src/tests/shakemap/coremods/dyfi_test.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
../opt/anaconda3/envs/shakemap/lib/python3.8/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests/shakemap/coremods/dyfi_test.py:9: in <module>
    from shakemap.coremods.dyfi import _get_dyfi_dataframe
shakemap/coremods/dyfi.py:9: in <module>
    from impactutils.io.table import dataframe_to_xml
E   ModuleNotFoundError: No module named 'impactutils.io.table'
-------- generated xml file: /Users/liliye/shakemap_src/test-output.xml --------
=========================== short test summary info ============================
ERROR tests/shakemap/coremods/dyfi_test.py
!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!
=============================== 1 error in 3.62s ===============================
(shakemap) liliye@liliyes-MacBook-Air shakemap_src % conda info -envs
usage: conda info [-h] [--json] [-v] [-q] [-a] [--base] [-e] [-s]
                  [--unsafe-channels]
conda info: error: argument -e/--envs: ignored explicit argument 'nvs'
(shakemap) liliye@liliyes-MacBook-Air shakemap_src % conda info --envs
# conda environments:
#
base                     /Users/liliye/opt/anaconda3
gmprocess                /Users/liliye/opt/anaconda3/envs/gmprocess
shakemap              *  /Users/liliye/opt/anaconda3/envs/shakemap

**Comment by emthompson-usgs** [comment]
I see what happened now. There was a change to one of the shakemap dependencies. I'll look into fixing it.

**Comment by emthompson-usgs** [comment]
Ok, it is fixed in PR #1115. If you don't want to wait for that to finish building, should be table to just do this and then run tests:
```
conda install impactutils=0.8.24
```

**Comment by cbworden** [comment]
Thanks, Eric, I just merged your PR. Mike and I are working on a full fix that should be ready today. We'll have to remember to un-pin Impactutils at that point. The reason we made the changes was that 1) it was difficult to get Pandas to use the right backend for xls/xlsx files and tests were failing as a result, and 2) we're moving all of the table-of-amps to JSON functions to the new mmitools repo. This should help reduce ShakeMap dependencies and therefore its sensitivity to changes in random packages. I'll comment here when the full fix is in.
